### DISCUSSION TITLE
Can ImageSharp take advantage of dedicated GPU?

### DISCUSSION THREAD

**Discussion by AlessandroMartinelli** [SixLabors/ImageSharp#2374] [Q&A]
As per title.

By taking a look at issues, it seems the answer is yes. However I'd like to know up to what extent is the gpu used by the library.

Thanks

**Comment by aloksharma1** [comment]
> As per title.
> 
> By taking a look at issues, it seems the answer is yes. However I'd like to know up to what extent is the gpu used by the library.
> 
> Thanks

did you find the answer to it, i need realtime image manipulation imagesharp is cross platform hence i am very eager to know this myself.

**Reply by AlessandroMartinelli** [reply]
@aloksharma1 sadly no, I never found an answer to it, thus I'm not using ImageSharp.

**Reply by aloksharma1** [reply]
whats the alternative? i found computesharp any other easy to use library like imagesharp?

**Reply by antonfirsov** [reply]
ComputeSharp has a very different purpose: it enables you writing shaders in C# on Windows. ImageSharp is a cross-platform general-purpose image-processing library.

**Comment by antonfirsov** [comment]
ImageSharp is CPU-only. I'm wondering what are your use-cases?

**Reply by aloksharma1** [reply]
my use case is real-time video feed processing, the video computations are done on cuda side then getting it on ImageSharp makes it loose performance due to cpu->gpu->cpu transitions of data this made me loose around half my fps on feed, anyways i also found that SkiaSharp can use Gpu so i have started implementing that now.

**Reply by antonfirsov** [reply]
> real-time video feed processing

Feels like this needs a video processing library rather than an image processing one. Have you considered FFmpeg? Looks like there is a [.NET wrapper](https://github.com/cmxl/FFmpeg.NET).

**Reply by aloksharma1** [reply]
my impression of ffmpeg was a video encoder decoder, can it also draw on frames? i need to show detection boxes and some other info on frames.
